diff --git a/lib/index.js b/lib/index.js
index ab0e21c0f2408c7790ba08fc9696dcda1ccd42ce..9922b11f3b76e663d38a3abfac82b12d17b59731 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -90,9 +90,9 @@ var originalSchema = tableNodes({
 	cellContent: "paragraph",
 	cellAttributes: { alignment: {
 		default: "left",
-		getFromDOM: (dom) => dom.style.textAlign || "left",
+		getFromDOM: (dom) => dom.dataset.proofTextAlign || dom.style.textAlign || "left",
 		setDOMAttr: (value, attrs) => {
-			attrs.style = `text-align: ${value || "left"}`;
+			attrs["data-proof-text-align"] = value || "left";
 		}
 	} }
 });
@@ -565,7 +565,10 @@ var tablePasteRule = $pasteRule((ctx) => ({ run: (slice, _view, isPlainText) =>
 	if (isPlainText) return slice;
 	function fixTable(node) {
 		const rowsCount = node.childCount;
-		const colsCount = node.lastChild?.childCount ?? 0;
+		let colsCount = 0;
+		node.forEach((row) => {
+			colsCount = Math.max(colsCount, row.childCount);
+		});
 		if (rowsCount === 0 || colsCount === 0) return paragraphSchema.type(ctx).create();
 		const headerRow = node.firstChild;
 		if (!(colsCount > 0 && headerRow && headerRow.childCount === 0)) return node;
